lld/include/lld-c/LLDAsLibraryC.h

```diff
@@ -131,10 +131,22 @@ void LLVMGetUndefinedReferencesEraVM(LLVMMemoryBufferRef inBuffer,
                                      char ***factoryDepSymbols,
                                      uint64_t *numFactoryDepSymbols);
 
-/** Disposes an array with linker symbols returned by the
+/** Disposes an array with symbols returned by the
  *  LLVMGetUndefinedReferencesEraVM(). */
-void LLVMDisposeUndefinedReferencesEraVM(char *linkerSymbolNames[],
-                                         uint64_t numLinkerSymbols);
+void LLVMDisposeUndefinedReferencesEraVM(char *symbolNames[],
+                                         uint64_t numSymbols);
+
+/** Returns true if the \p inBuffer contains an ELF object file. */
+LLVMBool LLVMIsELFEVM(LLVMMemoryBufferRef inBuffer);
```

Comment:
Any reason why we need two of these?

Reply:
`LLVMIsELFEVM` and `LLVMIsELFEraVM` differ in the target machine check. But we can pass target machine as a param.